### PR TITLE
37 create quiz UI

### LINKED ISSUE #37: Create Quiz UI
As a user, I want to have a nice-looking screen when doing a quiz, in order to stay focused and relaxed during the quiz.

**Acceptance Criteria:**
- [x] A screen with a simple button to start a quiz is created @GeorgZs 
  - [x] When clicking this button the quiz state resets and quiz screen is shown @GeorgZs
- [x] A quiz screen is created @GeorgZs
  - [x] Showing the current question @GeorgZs
  - [x] Showing buttons for each answer, which when clicked updates the screen as follows: @GeorgZs
    - [x] Disable the answer buttons @GeorgZs
    - [x] Indicate your answer and the right answer @GeorgZs
    - [x] Show the explanation @GeorgZs
    - [x] Show a button to continue to the next question @GeorgZs
  - [x] Showing the current progress within this quiz @GeorgZs
  - [x] Allowing the user to quit at any time @Lancear
    - [x] Popup with are-you-sure message @Lancear
    - [x] Call the quiz state cancel quiz early method @Lancear
- [x] When clicking continue after answering the last question the following happens @IVidackovic 
  - [x] The result is stored using the quiz state method @IVidackovic 
  - [x] A results screen is shown @IVidackovic 
- [x] The results screen shows the overall quiz score @IVidackovic 
- [x] The quiz session displays a timer. @IVidackovic 
- [x] The result screen shows a detailed score overview @IVidackovic 
  - [x] Think question / category level, completely free for the assignee to choose how the detailed score overview will look @IVidackovic 
  - [x] Feedback is gathered from the stakeholders and there are no major complaints left ignored @IVidackovic 
- [x] The results screen allows to navigate back to the start quiz screen @IVidackovic 
- [x] If time runs out, the user is brought to the results screen and unanswered questions are marked incorrect @Lancear

### PR DESCRIPTION
**Pull request by GeorgZs**
Closes #37 

- create quiz UI
- countdown timer implemented
- use Database questions to populate quiz and store results
- switch to a results screen after exam finishes

### :ballot_box_with_check: Definition of Done checklist
- [x] Meaningful title and valuable description in pull request
- [x] Code change tested manually if not covered by automated tests
- [x] Pull request linked to one or more issues/stories
- [x] Acceptance criteria of the linked stories satisfied
- [x] Code change comprehended by those who approved it
- [x] State of target branch maintained or improved after the code change
- [x] Number in the title 😜

### CODE REVIEW

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/PracticeTheoryActivity.java [hunk 1]
```diff
@@ -59,5 +88,118 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
...
+                    timerDialog.cancel();
+                    MILLIS = 0;
+                }
+
+            }
+        });
+
+        /*
+            @Override
+            public void onClick(View view) {
```
Comment (Lancear) [inline SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/PracticeTheoryActivity.java]: Maybe remove commented out code

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/PracticeTheoryActivity.java [hunk 2]
```diff
@@ -59,5 +88,118 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
...
+        });
     }
+
+    public void goToQuiz(){
+        Button button = findViewById(R.id.practiceQuiz);
+        button.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                //go to quiz screen
+                //TODO: start the Quiz State
```
Comment (Lancear) [inline SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/PracticeTheoryActivity.java]: Complete / Remove todo

File: SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/PracticeTheoryActivity.java [hunk 3]
```diff
@@ -3,26 +3,55 @@
...
 public class PracticeTheoryActivity extends AppCompatActivity {
 
     private BottomNavigationView bottomNavigationView;
+    private Dialog settingsDialog;
+    private ImageView settingsButton;
+
+    private int questionCount = 0;
+
+    //Countdown timer
+    public static final int TEN_MIN_IN_MILLIS = 10000;
```
Comment (Lancear) [inline SmartCarMQTTApp/app/src/main/java/com/example/smartcarmqttapp/PracticeTheoryActivity.java]: This is ten seconds not ten minutes, I seem to have forgotten to change it back after testing

**Review by Lancear — CHANGES_REQUESTED** [review]

**Review by Lancear — APPROVED** [review]
image

**Review by IVidackovic — APPROVED** [review]

**Pull request merged**